### ISSUE TITLE
Accessibility issue with Payment Item field

### ISSUE THREAD

**Issue by patric-boehner** [fluentform/fluentform#403]
The current setup for the payment item field is not accessible.

**Field Lable**
The label for the hidden product field type has no 'for' attribute connecting the label to the form field. As far as I am aware, having a form input with type="hidden" and a label without a corresponding "for" is an issue. There is no relationship between the label and the form field, which is confusing for form users using a screen reader. In voiceover, the payment item is skipped altogether when tabbing.

To address this, the label for the field should have a corresponding for attribute that matches the id of the input element it labels. The aria-label seems redundant at that point but there may be another reason for that.

**Updated Example:**
`<label for="ff_3_beef_quarter_side_product" aria-label="Beef Quarter Side">Beef Quarter Side </label>
<input type="hidden" id="ff_3_beef_quarter_side_product" name="beef_quarter_side_product" value="500">`

**Field Price Text**
The payment item field does provide more text that is important to a user in the "item price wrapper". This gives you the price label and the product price. However, the lack of association between the label and the span containing the price information could present an issue. Understanding the relationship between the label describing the product and the price information contained within the span is important, and right now, it's missing.

With the label issue fixes, a screen reader would just say "Beef Quarter Side" without mentioning the price.

Using an aria-label or aria-labeled on the pricing wrapper would be a possible fix.

**Updated Example:**
`<label for="ff_3_beef_quarter_side_product" aria-label="Beef Quarter Side">Beef Quarter Side</label>
<input type="hidden" id="ff_3_beef_quarter_side_product" aria-labelledby="product description">
<span id="product-description">
<span class="ff_product_price_label">Deposit Price:</span>
<span class="ff_product_price">$500.00</span>
</span>`
